docs/Integration_with_GitHub.rst

```diff
@@ -121,3 +276,361 @@ in that same pull request for netCDF, WRF, ...::
+and to maintain consistency across easyconfig files where desired.
+
+The ``--review-pr`` output consists of a 'multidiff' view per easyconfig file that is being touched by
+the specified pull request. The exact format of the output depends on whether EasyBuild is configured to allow
+colored output (enabled by default, see ``--color``).
+
+Search criteria for similar easyconfigs
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The set of existing similar easyconfig files is determined by specific search criteria; the first one that results
+in a non-empty set of easyconfigs is retained.
+
+The search criteria consists of a combination of the *software version criterion* with additional restrictions.
+
+The software version criterion is one of the criterions below (considered in order), with ``x.y.z`` the software
+version of the easyconfig file from the pull request:
+
+* exact same software version
+* same major/minor software version (same ``x`` and ``y``)
+* same major software version (same ``x``)
+* no (partial) version match (so consider any version)
+
+The addition restrictions are the following (also considered in order):
+
+* matching versionsuffix and toolchain name/version
+* matching versionsuffix and toolchain name (any toolchain version)
+* matching versionsuffix (any toolchain name/version)
+* matching toolchain name/version (any versionsuffix)
+* matching toolchain name (any versionsuffix, toolchain version)
+* no extra requirements (any versionsuffix, toolchain name/version)
+
+.. _github_new_update_pr:
+
+Submitting new and updating pull requests (``--new-pr``, ``--update-pr``)
+-------------------------------------------------------------------------
+
+*(supported since EasyBuild v2.6.0)*
+
+EasyBuild provides two simple yet powerful features that make contributing back to the central EasyBuild repositories
+significantly easier and less error-prone, especially for people who are not very familiar with ``git`` and/or GitHub
+yet:
+
+* ``--new-pr`` to create new pull requests
+* ``--update-pr`` to update existing pull requests
+
+.. note:: Both ``--new-pr`` and ``--update-pr`` are still **experimental**, meaning that their behaviour may change in
+          future releases and that they require having ``--experimental`` enabled;
+          see also :ref:`experimental_features`.
+
+.. _github_new_pr:
+
+Submitting pull requests (``--new-pr``)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To create a new pull request, the ``--new-pr`` command line option can be used, provided that the necessary
+requirements are fullfilled (see :ref:`github_requirements`).
+
+In its simplest form, you just provide the location of the file(s) that you want to include in the pull request::
+
+    $ eb --new-pr test.eb --experimental
+
+This takes care of all the steps required to make a contribution, i.e.:
+
+* set up a working copy of the relevant EasyBuild repository (e.g., ``easybuild-easyconfigs``)
```

Comment:
This is a bit unclear to me.  From the example, it seems that only easyconfigs can be handles by this option.  But with the information about `--pr-target-repo` below, I get the impression that also framework/easyblock PRs can be handled.  However, in this case, the question is how the automatic renaming/moving works.